contracts/sg721-base/src/contract.rs

```diff
@@ -214,6 +221,82 @@ where
         Ok(res)
     }
 
+    pub fn update_collection_info(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        new_collection: CollectionInfo<RoyaltyInfoResponse>,
+    ) -> Result<Response, ContractError> {
+        let frozen_collection_info = self.frozen_collection_info.load(deps.storage)?;
+
+        if frozen_collection_info {
+            return Err(ContractError::CollectionInfoFrozen {});
+        }
+
+        let collection = self.query_collection_info(deps.as_ref())?;
+        if collection.creator != info.sender {
+            return Err(ContractError::Unauthorized {});
+        }
+
+        deps.api.addr_validate(&new_collection.creator)?;
+
+        if new_collection.description.len() > MAX_DESCRIPTION_LENGTH as usize {
+            return Err(ContractError::DescriptionTooLong {});
+        }
+
+        let image = Url::parse(&new_collection.image)?;
+
+        if let Some(ref external_link) = new_collection.external_link {
+            Url::parse(external_link)?;
+        }
+
+        if let Some(start_trading_time) = new_collection.start_trading_time {
+            if env.block.time > start_trading_time {
+                return Err(ContractError::InvalidStartTradingTime {});
+            }
+        }
```

Comment:
start trading time should only be be updated  through the minter because it needs to check public mint start time + offset, also it should only allow to decrease it after public mint starts and this will allow to just keep extending it until is frozen.